Add setTypingStatus(boolean) to IUser

ArsenArsen opened the pull request:
### Prerequisites
* [x] This pull request follows the code style of the project
* [x] I have read and understand the [contribution guidelines](CONTRIBUTING.md)
* [x] I have tested this feature thoroughly

**Issues Fixed:** Added a setTypingStatus method, which was missing

### Changes Proposed in this Pull Request
* Added the setTypingStatus(boolean)
* Renamed the old internal setTypingStatus into SetCachedTypingStatus

...

austinv11 commented on src/main/java/sx/blah/discord/handle/impl/obj/Channel.java:
```diff
@@ -270,7 +270,12 @@ public IInvite createInvite(int maxAge, int maxUses, boolean temporary) throws M
+	public void setTypingStatus(boolean typing) {
+		isTyping.set(typing);
 
 		typingTimer.scheduleAtFixedRate(new TimerTask() {
```
You should only schedule the task if typing = true

austinv11 commented on src/main/java/sx/blah/discord/handle/impl/obj/Channel.java:
```diff
@@ -579,7 +584,7 @@ public boolean equals(Object other) {
 	 * @param typingStatus The new typing status.
 	 */
-	public void setTypingStatus(boolean typingStatus) {
+	public void setCachedTypingStatus(boolean typingStatus) {
```
You could probably get away with removing this method altogether as its mirrored in line 278

ArsenArsen commented:
@GrandPanda @austinv11 I have done the things noted above. I also removed the setCachedTypingStatus method and do setTypingStatus(false) instead in DiscordWS.java I removed those exceptions from the docs and tested, and I can confirm it is working as expected.
